src/mus1/core/utils/file_hash.py:58: error: Incompatible types in assignment (expression has type "blake2b", variable has type "HASH")  [assignment]
src/mus1/core/config_manager.py:380: error: Need type annotation for "scope_data" (hint: "scope_data: dict[<type>, <type>] = ...")  [var-annotated]
src/mus1/core/config_manager.py:637: error: Need type annotation for "result" (hint: "result: dict[<type>, <type>] = ...")  [var-annotated]
src/mus1/core/scanners/base_scanner.py:15: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
src/mus1/core/scanners/base_scanner.py:16: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
src/mus1/core/logging_bus.py:41: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
src/mus1/core/logging_bus.py:141: error: Cannot infer type of lambda  [misc]
src/mus1/core/job_provider.py:13: error: Module "mus1.core.metadata" has no attribute "WorkerEntry"  [attr-defined]
src/mus1/core/job_provider.py:84: error: Incompatible types in assignment (expression has type "Popen[str]", variable has type "CompletedProcess[str]")  [assignment]
src/mus1/core/job_provider.py:113: error: "CompletedProcess[str]" has no attribute "poll"  [attr-defined]
src/mus1/core/job_provider.py:117: error: "CompletedProcess[str]" has no attribute "kill"  [attr-defined]
src/mus1/core/job_provider.py:118: error: "CompletedProcess[str]" has no attribute "wait"  [attr-defined]
src/mus1/core/job_provider.py:232: error: Incompatible types in assignment (expression has type "Popen[str]", variable has type "CompletedProcess[str]")  [assignment]
src/mus1/core/job_provider.py:255: error: "CompletedProcess[str]" has no attribute "wait"  [attr-defined]
src/mus1/core/job_provider.py:307: error: Incompatible types in assignment (expression has type "Popen[str]", variable has type "CompletedProcess[str]")  [assignment]
src/mus1/core/job_provider.py:330: error: "CompletedProcess[str]" has no attribute "wait"  [attr-defined]
src/mus1/core/scanners/remote.py:61: error: Name "StateManager" is not defined  [name-defined]
src/mus1/core/scanners/remote.py:62: error: Name "DataManager" is not defined  [name-defined]
src/mus1/core/scanners/remote.py:107: error: Name "StateManager" is not defined  [name-defined]
src/mus1/core/scanners/remote.py:108: error: Name "DataManager" is not defined  [name-defined]
src/mus1/core/scanners/remote.py:136: error: Name "StateManager" is not defined  [name-defined]
src/mus1/core/scanners/remote.py:137: error: Name "DataManager" is not defined  [name-defined]
src/mus1/core/schema.py:22: error: Variable "mus1.core.schema.Base" is not valid as a type  [valid-type]
src/mus1/core/schema.py:22: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/mus1/core/schema.py:22: error: Invalid base class "Base"  [misc]
src/mus1/core/schema.py:38: error: Variable "mus1.core.schema.Base" is not valid as a type  [valid-type]
src/mus1/core/schema.py:38: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/mus1/core/schema.py:38: error: Invalid base class "Base"  [misc]
src/mus1/core/schema.py:44: error: Need type annotation for "sex"  [var-annotated]
src/mus1/core/schema.py:45: error: Need type annotation for "designation"  [var-annotated]
src/mus1/core/schema.py:57: error: Variable "mus1.core.schema.Base" is not valid as a type  [valid-type]
src/mus1/core/schema.py:57: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/mus1/core/schema.py:57: error: Invalid base class "Base"  [misc]
src/mus1/core/schema.py:65: error: Need type annotation for "processing_stage"  [var-annotated]
src/mus1/core/schema.py:74: error: Variable "mus1.core.schema.Base" is not valid as a type  [valid-type]
src/mus1/core/schema.py:74: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/mus1/core/schema.py:74: error: Invalid base class "Base"  [misc]
src/mus1/core/schema.py:95: error: Variable "mus1.core.schema.Base" is not valid as a type  [valid-type]
src/mus1/core/schema.py:95: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/mus1/core/schema.py:95: error: Invalid base class "Base"  [misc]
src/mus1/core/schema.py:122: error: Variable "mus1.core.schema.Base" is not valid as a type  [valid-type]
src/mus1/core/schema.py:122: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/mus1/core/schema.py:122: error: Invalid base class "Base"  [misc]
src/mus1/core/schema.py:130: error: Need type annotation for "provider"  [var-annotated]
src/mus1/core/schema.py:133: error: Variable "mus1.core.schema.Base" is not valid as a type  [valid-type]
src/mus1/core/schema.py:133: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/mus1/core/schema.py:133: error: Invalid base class "Base"  [misc]
src/mus1/core/schema.py:139: error: Need type annotation for "kind"  [var-annotated]
src/mus1/core/schema.py:143: error: Variable "mus1.core.schema.Base" is not valid as a type  [valid-type]
src/mus1/core/schema.py:143: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/mus1/core/schema.py:143: error: Invalid base class "Base"  [misc]
src/mus1/core/schema.py:154: error: Variable "mus1.core.schema.Base" is not valid as a type  [valid-type]
src/mus1/core/schema.py:154: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/mus1/core/schema.py:154: error: Invalid base class "Base"  [misc]
src/mus1/core/schema.py:165: error: Variable "mus1.core.schema.Base" is not valid as a type  [valid-type]
src/mus1/core/schema.py:165: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/mus1/core/schema.py:165: error: Invalid base class "Base"  [misc]
src/mus1/core/schema.py:180: error: Variable "mus1.core.schema.Base" is not valid as a type  [valid-type]
src/mus1/core/schema.py:180: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/mus1/core/schema.py:180: error: Invalid base class "Base"  [misc]
src/mus1/core/schema.py:196: error: Variable "mus1.core.schema.Base" is not valid as a type  [valid-type]
src/mus1/core/schema.py:196: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/mus1/core/schema.py:196: error: Invalid base class "Base"  [misc]
src/mus1/core/schema.py:211: error: Variable "mus1.core.schema.Base" is not valid as a type  [valid-type]
src/mus1/core/schema.py:211: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/mus1/core/schema.py:211: error: Invalid base class "Base"  [misc]
src/mus1/core/schema.py:224: error: Variable "mus1.core.schema.Base" is not valid as a type  [valid-type]
src/mus1/core/schema.py:224: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/mus1/core/schema.py:224: error: Invalid base class "Base"  [misc]
src/mus1/core/schema.py:233: error: Variable "mus1.core.schema.Base" is not valid as a type  [valid-type]
src/mus1/core/schema.py:233: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/mus1/core/schema.py:233: error: Invalid base class "Base"  [misc]
src/mus1/core/schema.py:242: error: Variable "mus1.core.schema.Base" is not valid as a type  [valid-type]
src/mus1/core/schema.py:242: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/mus1/core/schema.py:242: error: Invalid base class "Base"  [misc]
src/mus1/core/schema.py:249: error: Variable "mus1.core.schema.Base" is not valid as a type  [valid-type]
src/mus1/core/schema.py:249: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/mus1/core/schema.py:249: error: Invalid base class "Base"  [misc]
src/mus1/core/schema.py:261: error: Variable "mus1.core.schema.Base" is not valid as a type  [valid-type]
src/mus1/core/schema.py:261: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/mus1/core/schema.py:261: error: Invalid base class "Base"  [misc]
src/mus1/core/schema.py:314: error: Name "Colony" is not defined  [name-defined]
src/mus1/core/schema.py:343: error: Name "Subject" is not defined  [name-defined]
src/mus1/core/schema.py:372: error: Name "Experiment" is not defined  [name-defined]
src/mus1/core/schema.py:396: error: Name "PluginMetadata" is not defined  [name-defined]
src/mus1/core/schema.py:421: error: Name "PluginResult" is not defined  [name-defined]
src/mus1/core/schema.py:453: error: Name "User" is not defined  [name-defined]
src/mus1/core/schema.py:479: error: Name "Lab" is not defined  [name-defined]
src/mus1/core/schema.py:500: error: Name "Workgroup" is not defined  [name-defined]
src/mus1/core/repository.py:34: error: Name "Colony" is not defined  [name-defined]
src/mus1/core/repository.py:43: error: Name "Colony" is not defined  [name-defined]
src/mus1/core/repository.py:51: error: Name "Colony" is not defined  [name-defined]
src/mus1/core/repository.py:59: error: Name "Colony" is not defined  [name-defined]
src/mus1/core/repository.py:247: error: "Result[Any]" has no attribute "rowcount"  [attr-defined]
src/mus1/core/repository.py:282: error: Incompatible types in assignment (expression has type "str", variable has type "Column[str]")  [assignment]
src/mus1/core/repository.py:283: error: Incompatible types in assignment (expression has type "datetime | None", variable has type "Column[datetime]")  [assignment]
src/mus1/core/repository.py:284: error: Incompatible types in assignment (expression has type "int", variable has type "Column[int]")  [assignment]
src/mus1/core/repository.py:285: error: Incompatible types in assignment (expression has type "float", variable has type "Column[float]")  [assignment]
src/mus1/core/repository.py:286: error: Incompatible types in assignment (expression has type "datetime", variable has type "Column[datetime]")  [assignment]
src/mus1/core/repository.py:291: error: Argument "hash" to "VideoFile" has incompatible type "Column[str]"; expected "str"  [arg-type]
src/mus1/core/repository.py:292: error: Argument "recorded_time" to "VideoFile" has incompatible type "Column[datetime]"; expected "datetime | None"  [arg-type]
src/mus1/core/repository.py:293: error: Argument "size_bytes" to "VideoFile" has incompatible type "Column[int]"; expected "int"  [arg-type]
src/mus1/core/repository.py:294: error: Argument "last_modified" to "VideoFile" has incompatible type "Column[float]"; expected "float"  [arg-type]
src/mus1/core/repository.py:295: error: Argument "date_added" to "VideoFile" has incompatible type "Column[datetime]"; expected "datetime"  [arg-type]
src/mus1/core/repository.py:312: error: Argument "hash" to "VideoFile" has incompatible type "Column[str]"; expected "str"  [arg-type]
src/mus1/core/repository.py:313: error: Argument "recorded_time" to "VideoFile" has incompatible type "Column[datetime]"; expected "datetime | None"  [arg-type]
src/mus1/core/repository.py:314: error: Argument "size_bytes" to "VideoFile" has incompatible type "Column[int]"; expected "int"  [arg-type]
src/mus1/core/repository.py:315: error: Argument "last_modified" to "VideoFile" has incompatible type "Column[float]"; expected "float"  [arg-type]
src/mus1/core/repository.py:316: error: Argument "date_added" to "VideoFile" has incompatible type "Column[datetime]"; expected "datetime"  [arg-type]
src/mus1/core/repository.py:328: error: Argument "hash" to "VideoFile" has incompatible type "Column[str]"; expected "str"  [arg-type]
src/mus1/core/repository.py:329: error: Argument "recorded_time" to "VideoFile" has incompatible type "Column[datetime]"; expected "datetime | None"  [arg-type]
src/mus1/core/repository.py:330: error: Argument "size_bytes" to "VideoFile" has incompatible type "Column[int]"; expected "int"  [arg-type]
src/mus1/core/repository.py:331: error: Argument "last_modified" to "VideoFile" has incompatible type "Column[float]"; expected "float"  [arg-type]
src/mus1/core/repository.py:332: error: Argument "date_added" to "VideoFile" has incompatible type "Column[datetime]"; expected "datetime"  [arg-type]
src/mus1/core/repository.py:345: error: Argument "hash" to "VideoFile" has incompatible type "Column[str]"; expected "str"  [arg-type]
src/mus1/core/repository.py:346: error: Argument "recorded_time" to "VideoFile" has incompatible type "Column[datetime]"; expected "datetime | None"  [arg-type]
src/mus1/core/repository.py:347: error: Argument "size_bytes" to "VideoFile" has incompatible type "Column[int]"; expected "int"  [arg-type]
src/mus1/core/repository.py:348: error: Argument "last_modified" to "VideoFile" has incompatible type "Column[float]"; expected "float"  [arg-type]
src/mus1/core/repository.py:349: error: Argument "date_added" to "VideoFile" has incompatible type "Column[datetime]"; expected "datetime"  [arg-type]
src/mus1/core/repository.py:396: error: Argument "name" to "Worker" has incompatible type "Column[str]"; expected "str"  [arg-type]
src/mus1/core/repository.py:397: error: Argument "ssh_alias" to "Worker" has incompatible type "Column[str]"; expected "str"  [arg-type]
src/mus1/core/repository.py:398: error: Argument "role" to "Worker" has incompatible type "Column[str]"; expected "str | None"  [arg-type]
src/mus1/core/repository.py:400: error: Argument "os_type" to "Worker" has incompatible type "Column[str]"; expected "str | None"  [arg-type]
src/mus1/core/repository.py:412: error: Argument "name" to "Worker" has incompatible type "Column[str]"; expected "str"  [arg-type]
src/mus1/core/repository.py:413: error: Argument "ssh_alias" to "Worker" has incompatible type "Column[str]"; expected "str"  [arg-type]
src/mus1/core/repository.py:414: error: Argument "role" to "Worker" has incompatible type "Column[str]"; expected "str | None"  [arg-type]
src/mus1/core/repository.py:416: error: Argument "os_type" to "Worker" has incompatible type "Column[str]"; expected "str | None"  [arg-type]
src/mus1/core/repository.py:428: error: Argument "name" to "Worker" has incompatible type "Column[str]"; expected "str"  [arg-type]
src/mus1/core/repository.py:429: error: Argument "ssh_alias" to "Worker" has incompatible type "Column[str]"; expected "str"  [arg-type]
src/mus1/core/repository.py:430: error: Argument "role" to "Worker" has incompatible type "Column[str]"; expected "str | None"  [arg-type]
src/mus1/core/repository.py:432: error: Argument "os_type" to "Worker" has incompatible type "Column[str]"; expected "str | None"  [arg-type]
src/mus1/core/repository.py:453: error: Argument "name" to "ScanTarget" has incompatible type "Column[str]"; expected "str"  [arg-type]
src/mus1/core/repository.py:455: error: Argument 1 to "loads" has incompatible type "Column[str]"; expected "str | bytes | bytearray"  [arg-type]
src/mus1/core/repository.py:456: error: Argument "ssh_alias" to "ScanTarget" has incompatible type "Column[str]"; expected "str | None"  [arg-type]
src/mus1/core/repository.py:469: error: Argument "name" to "ScanTarget" has incompatible type "Column[str]"; expected "str"  [arg-type]
src/mus1/core/repository.py:471: error: Argument 1 to "loads" has incompatible type "Column[str]"; expected "str | bytes | bytearray"  [arg-type]
src/mus1/core/repository.py:472: error: Argument "ssh_alias" to "ScanTarget" has incompatible type "Column[str]"; expected "str | None"  [arg-type]
src/mus1/core/repository.py:485: error: Argument "name" to "ScanTarget" has incompatible type "Column[str]"; expected "str"  [arg-type]
src/mus1/core/repository.py:487: error: Argument 1 to "loads" has incompatible type "Column[str]"; expected "str | bytes | bytearray"  [arg-type]
src/mus1/core/repository.py:488: error: Argument "ssh_alias" to "ScanTarget" has incompatible type "Column[str]"; expected "str | None"  [arg-type]
src/mus1/core/repository.py:495: error: Name "User" is not defined  [name-defined]
src/mus1/core/repository.py:505: error: Name "User" is not defined  [name-defined]
src/mus1/core/repository.py:514: error: Name "User" is not defined  [name-defined]
src/mus1/core/repository.py:523: error: Name "User" is not defined  [name-defined]
src/mus1/core/repository.py:533: error: Name "Lab" is not defined  [name-defined]
src/mus1/core/repository.py:543: error: Name "Lab" is not defined  [name-defined]
src/mus1/core/repository.py:552: error: Name "Lab" is not defined  [name-defined]
src/mus1/core/repository.py:561: error: Name "Lab" is not defined  [name-defined]
src/mus1/core/repository.py:637: error: Name "Lab" is not defined  [name-defined]
src/mus1/core/repository.py:685: error: Argument "name" to "Worker" has incompatible type "Column[str]"; expected "str"  [arg-type]
src/mus1/core/repository.py:686: error: Argument "ssh_alias" to "Worker" has incompatible type "Column[str]"; expected "str"  [arg-type]
src/mus1/core/repository.py:687: error: Argument "role" to "Worker" has incompatible type "Column[str]"; expected "str | None"  [arg-type]
src/mus1/core/repository.py:689: error: Argument "os_type" to "Worker" has incompatible type "Column[str]"; expected "str | None"  [arg-type]
src/mus1/core/repository.py:722: error: Argument "name" to "ScanTarget" has incompatible type "Column[str]"; expected "str"  [arg-type]
src/mus1/core/repository.py:724: error: Argument 1 to "loads" has incompatible type "Column[str]"; expected "str | bytes | bytearray"  [arg-type]
src/mus1/core/repository.py:725: error: Argument "ssh_alias" to "ScanTarget" has incompatible type "Column[str]"; expected "str | None"  [arg-type]
src/mus1/core/setup_service.py:117: error: Incompatible types in assignment (expression has type "None", variable has type "list[str]")  [assignment]
src/mus1/core/setup_service.py:299: error: Name "result" is not defined  [name-defined]
src/mus1/core/setup_service.py:303: error: Name "result" is not defined  [name-defined]
src/mus1/core/setup_service.py:1343: error: "object" has no attribute "append"  [attr-defined]
src/mus1/core/setup_service.py:1345: error: "object" has no attribute "append"  [attr-defined]
src/mus1/core/setup_service.py:1351: error: "object" has no attribute "append"  [attr-defined]
src/mus1/core/setup_service.py:1353: error: "object" has no attribute "append"  [attr-defined]
src/mus1/core/setup_service.py:1359: error: "object" has no attribute "append"  [attr-defined]
src/mus1/core/setup_service.py:1361: error: "object" has no attribute "append"  [attr-defined]
src/mus1/core/setup_service.py:1367: error: "object" has no attribute "append"  [attr-defined]
src/mus1/core/setup_service.py:1369: error: "object" has no attribute "append"  [attr-defined]
src/mus1/core/setup_service.py:1375: error: "object" has no attribute "append"  [attr-defined]
src/mus1/core/setup_service.py:1377: error: "object" has no attribute "append"  [attr-defined]
src/mus1/core/setup_service.py:1381: error: "object" has no attribute "append"  [attr-defined]
src/mus1/core/project_manager_clean.py:133: error: Argument 1 to "get_colony" of "ProjectManagerClean" has incompatible type "str | None"; expected "str"  [arg-type]
src/mus1/core/project_manager_clean.py:588: error: "VideoRepository" has no attribute "find_all"  [attr-defined]
src/mus1/core/project_manager_clean.py:617: error: Incompatible default for argument "batch_name" (default has type "None", argument has type "str")  [assignment]
src/mus1/core/project_manager_clean.py:617: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
src/mus1/core/project_manager_clean.py:617: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
src/mus1/core/project_manager_clean.py:617: error: Incompatible default for argument "description" (default has type "None", argument has type "str")  [assignment]
src/mus1/core/project_manager_clean.py:617: error: Incompatible default for argument "selection_criteria" (default has type "None", argument has type "dict[str, Any]")  [assignment]
src/mus1/core/plugin_manager_clean.py:168: error: "EntryPoints" has no attribute "get"  [attr-defined]
src/mus1/core/plugin_manager_clean.py:260: error: Incompatible default for argument "sort_mode" (default has type "None", argument has type "str")  [assignment]
src/mus1/core/plugin_manager_clean.py:260: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
src/mus1/core/plugin_manager_clean.py:260: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
src/mus1/core/simple_cli.py:587: error: Item "None" of "UserProfileDTO | None" has no attribute "name"  [union-attr]
src/mus1/core/simple_cli.py:1037: error: Argument 1 to "create_lab" of "SetupService" has incompatible type "mus1.core.metadata.LabDTO"; expected "mus1.core.setup_service.LabDTO"  [arg-type]
src/mus1/core/simple_cli.py:1116: error: Argument 2 to "create_colony" of "SetupService" has incompatible type "mus1.core.metadata.ColonyDTO"; expected "mus1.core.setup_service.ColonyDTO"  [arg-type]
Found 159 errors in 11 files (checked 17 source files)
